pkg/api/deprecation_test.go

```diff
@@ -0,0 +1,229 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift-online/ocm-common/pkg/ocm/consts"
+)
+
+// ErrorTemplate represents a template that can be used to send an error.
+type ErrorTemplate struct {
+	// Status is the HTTP status of the response.
+	Status int
+
+	// ID is the numeric identifier of the error within the service.
+	ID int
+
+	// Reason is a function that generates a default reason message for the error from the HTTP
+	// request object. If this is nil then there will be no default error message.
+	Reason func(r *http.Request) string
+}
+
+// Format generates an error reason from the given format and arguments, creates an instance of the
+// error template and returns it. If the format string is empty, then the default reason of
+// the template will be used.
+func (t *ErrorTemplate) Format(r *http.Request, format string, a ...interface{}) Error {
+
+	// Generate the reason message:
+	var reason string
+	if format != "" {
+		reason = fmt.Sprintf(format, a...)
+	} else if t.Reason != nil {
+		reason = t.Reason(r)
+	}
+
+	// Prepare the body:
+	return Error{
+		ID:        fmt.Sprintf("%d", t.ID),
+		Code:      fmt.Sprintf("CLUSTERS-MGMT-%d", t.ID),
+		Reason:    reason,
+		Timestamp: time.Now().UTC(),
+	}
+}
+
+// E410 is a template for a generic HTTP `410 Gone` error.
+var E410 = &ErrorTemplate{
+	Status: http.StatusGone,
+	ID:     410,
+	Reason: func(r *http.Request) string {
+		return fmt.Sprintf(
+			"The requested resource '%s' is no longer available and will not be available again",
+			r.URL.Path,
+		)
+	},
+}
+
+var _ = Describe("Deprecation Middleware", func() {
+	var (
+		nextHandler http.Handler
+		handler     http.Handler
+		rr          *httptest.ResponseRecorder
```

Comment:
Avoiding 2 char variable names is helpful

Reply:
There are some conventions like index in iterations (i, j) or "ok" and "k"/"v" for key/value of a map. That said, for anything more complicated it make sense to provide some meaningful name which helps developers to maintain the code.

Aside from that, Go conventions are followed but not always - for example, abrrevations in Go are suppose to be written in capital. We dislike that because it creates unreadable terms like AWSVPCARN (awsVpcArn is easier), so we deviated to lowerCamelCase.

Reply:
Exactly! And I totally agree with you.